onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -2862,17 +2863,8 @@ ORT_API_STATUS_IMPL(OrtApis::Node_GetAttributeByName, _In_ const OrtNode* node,
     return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "'attribute' argument is NULL");
   }
 
-  std::unique_ptr<OrtArrayOfConstObjects> array;
-  ORT_API_RETURN_IF_STATUS_NOT_OK(node->GetAttributes(array));
-
-  *attribute = nullptr;
-  for (auto attr : array->storage) {
-    auto ort_op_attr = reinterpret_cast<const OrtOpAttr*>(attr);
-    if (ort_op_attr->attr_proto.has_name() && (ort_op_attr->attr_proto.name() == attribute_name)) {
-      *attribute = ort_op_attr;
-      break;
-    }
-  }
+  const EpNode* ep_node = EpNode::ToInternal(node);
```

Comment:
One other thing here. `EpNode::ToInternal` returns `nullptr` if this is not an `EpNode` (it could be a `ModelEditorNode`). In that case, we should return a status error.

Reply:
Ah, right, just addressed this.